pkg/controller/cassandracluster/cassandra_status.go

```diff
@@ -405,13 +424,32 @@ func (rcc *ReconcileCassandraCluster) UpdateCassandraRackStatusPhase(cc *api.Cas
 
 	if status.CassandraRackStatus[dcRackName].Phase == api.ClusterPhaseInitial {
 
+		nodesPerRacks := cc.GetNodesPerRacks(dcRackName)
+		//If we are stuck in initializing state, we can rollback the add of dc which implies decommissioning nodes
+		if nodesPerRacks <=0 {
+			logrus.WithFields(logrus.Fields{"cluster": cc.Name,
+				"rack": dcRackName}).Warn("Aborting Initializing..., start ScaleDown")
+			status.CassandraRackStatus[dcRackName].Phase = api.ClusterPhasePending
+			now := metav1.Now()
+			lastAction.StartTime = &now
+			lastAction.Status = api.StatusToDo
+			lastAction.Name = api.ActionScaleDown
+			status.CassandraRackStatus[dcRackName].PodLastOperation.Status = api.StatusToDo
+			status.CassandraRackStatus[dcRackName].PodLastOperation.Name = api.OperationDecommission
+			status.CassandraRackStatus[dcRackName].PodLastOperation.StartTime = &now
```

Comment:
I think we set all the following parameters in different places the same way. We should probably refactor that code